server/modlog.ts

```diff
@@ -450,6 +450,26 @@ export class Modlog {
 		};
 	}
 
+	hotpatch() {
+		if (this !== Rooms.Modlog) {
+			throw new Error(`Attempt to call Modlog#hotpatch on a modlog instance other than Rooms.Modlog`);
+		}
+
+		const streams = Rooms.Modlog.streams;
+		const sharedStreams = Rooms.Modlog.sharedStreams;
+
+		const processManagers = ProcessManager.processManagers;
+		for (const manager of processManagers.slice()) {
+			if (manager.filename.startsWith(__filename)) void manager.destroy();
+		}
```

Comment:
Why the manual search for PMs, anyway? I believe our usual solution to this problem is to export `Rooms.Modlog.PM`.

Reply:
Chat plugins search manually; I can certainly just put the PM in `Rooms` if thats better.

Reply:
I might not be understanding correctly, so correct me if I'm wrong: You're restarting chat-plugin processes but only for chat plugins that use the Modlog library?

That seems really fragile/buggy. I would recommend just always hotpatching chat when hotpatching modlog (like how hotpatching formats also hotpatches battles).

Reply:
No, `/hotpatch chat` manually searches.

Reply:
Oh, it turns out I just merged hotpatching formats/battles a while ago. But the point is, I think you should just hotpatch chat; partially hotpatching chat sounds unnecessarily dangerous.

Reply:
If you want to hotpatch chat every time why not just make `Modlog` part of `Chat`